### PR TITLE
Task3 Greeting servlet

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
We visit a city by pressing the button. Once we visit that city that city is flagged in our session as visited so that next time we click to visit it a different greeting message appears.

### CODE REVIEW

**Comment by VasilMitovClouway** [comment]
@mgenov Pull request is ready for review.

File: src/test/java/servlets/GreetingServletTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  @Rule
+  public JUnitRuleMockery context = new JUnitRuleMockery();
+
+  private HttpServletRequest req = context.mock(HttpServletRequest.class);
+  private HttpServletResponse resp = context.mock(HttpServletResponse.class);
+  private FakeSession session = new FakeSession();
+
+  @Test
+  public void happyPath() throws Exception {
```
Comment (mgenov) [inline src/test/java/servlets/GreetingServletTest.java]: visitForTheFirstTime

File: src/test/java/servlets/GreetingServletTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }});
+
+    servlet.doGet(req, resp);
+    servlet.doPost(req, resp);
+
+    assertThat(out.toString(), containsString("Welcome this is your first time here."));
+  }
+
+  @Test
+  public void visitingAnAlreadyVisitedCity() throws Exception {
```
Comment (mgenov) [inline src/test/java/servlets/GreetingServletTest.java]: cityWasAlreadyVisited

File: src/main/java/servlets/GreetingServlet.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class GreetingServlet extends HttpServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String message="Welcome this is your first time here.";
+    HttpSession session=req.getSession();
+    if(session.getAttribute("message")==null){
```
Comment (mgenov) [inline src/main/java/servlets/GreetingServlet.java]: make sure that code is formatted cause it doesn't look so.

File: src/test/java/servlets/GreetingServletTest.java (hunk 1 above)
Comment (VasilMitovClouway) [inline src/test/java/servlets/GreetingServletTest.java]: Changed.

File: src/main/java/servlets/GreetingServlet.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import java.io.PrintWriter;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class GreetingServlet extends HttpServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String message = "Welcome this is your first time here.";
```
Comment (mgenov) [inline src/main/java/servlets/GreetingServlet.java]: could be constant for that class cause it's not modified.

**Comment by mgenov** [comment]
Please squash your changes so I can merge them after review completes.

File: src/main/java/servlets/GreetingServlet.java (hunk 4 above)
Comment (VasilMitovClouway) [inline src/main/java/servlets/GreetingServlet.java]: Changed.

File: src/main/java/servlets/GreetingServlet.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class GreetingServlet extends HttpServlet {
+  private String message = "Welcome this is your first time here.";
```
Comment (mgenov) [inline src/main/java/servlets/GreetingServlet.java]: private static final String message = "Welcome this is your first time here." is a constant.

**Pull request merged**